src/main/steps/c100-rebuild/address/manual/ManualAddressPostController.ts

```diff
@@ -0,0 +1,91 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import {
+  C100Address,
+  C100Applicant,
+  C100RebuildPartyDetails,
+  PartyType,
+  YesOrNo,
+} from '../../../../app/case/definition';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { Form, FormFields, FormFieldsFn } from '../../../../app/form/Form';
+import { PartyDetailsVariant, getPartyDetails, transformPartyDetails, updatePartyDetails } from '../../people/util';
+import { C100AddressForm, C100UrlPartyType } from '../definitions';
+import { getUpdatedForm } from '../utils';
+
+import { applicantForm as applicantManualAddressForm, form as manualAddressForm } from './address-manual';
+
+@autobind
+export default class ManualAddressPostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { id, partyType } = req.params;
+    const form = new Form(
+      getUpdatedForm(
+        req.session.userCase,
+        id,
+        partyType as C100UrlPartyType,
+        partyType === C100UrlPartyType.APPLICANT
+          ? (applicantManualAddressForm as C100AddressForm)
+          : (manualAddressForm as C100AddressForm),
+        null,
+        YesOrNo.YES
+      ).fields as FormFields
+    );
+    const { onlycontinue, saveAndComeLater, ...formFields } = req.body;
+    const { _csrf, ...formData } = form.getParsedBody(formFields);
+
+    req = this.updateAddressDetails(req, id, partyType as C100UrlPartyType, formData);
+
+    if (onlycontinue) {
+      req.session.errors = form.getErrors(formData);
+      return this.redirect(req, res);
+    } else if (saveAndComeLater) {
+      this.saveAndComeLater(req, res, req.session.userCase);
+    }
+  }
+
+  private updateAddressDetails = (req: AppRequest<AnyObject>, id: string, partyType: C100UrlPartyType, formData) => {
```

Comment:
return type missing

Reply:
i don't think you need to return anything

Reply:
removed the return statement in updateAddressDetails